discovery_engine_core/core/src/engine.rs

```diff
@@ -548,13 +549,31 @@ async fn update_stacks<'a>(
 
     let mut errors = Vec::new();
     for stack in &mut stacks {
+        // bail if breaking news failed and key phrases for personalized news are empty
+        macro_rules! bail_if_breaking_news_failed_without_key_phrases {
```

Comment:
how are we going to handle this once #304 introduced `TrustedNews` which also doesn't depend on key phrases? i guess we should only fail if all stacks which don't need key phrases failed and the key phrases are empty?